Source/Core/Core/State.cpp

```diff
@@ -239,7 +235,7 @@ void SaveToBuffer(Core::System& system, std::vector<u8>& buffer)
 
         DoState(system, p_measure);
         const size_t buffer_size = reinterpret_cast<size_t>(ptr);
-        buffer.resize(buffer_size);
+        buffer = Common::UniqueBuffer<u8>(buffer_size);
```

Comment:
To avoid the performance hit of reallocating, should we check if the buffer is already large enough? If not, it would make sense to return the buffer from this function instead of having it as a parameter.

Reply:
I think "malloc" and equivalents are a relatively cheap call, but I've changed it to only reallocate when the buffer grows.

Reply:
Yeah, compared to everything else we're doing, it's probably pretty cheap.